housekeeping/clang-tidy-diff.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -xe
+
+cd $(dirname $0)/..
+git diff -U0 HEAD^ | clang-tidy-diff.py -p1 -path $BUILD_DIR -regex "core/.*\.hpp"
```

Comment:
1. How this will work on master baranch?
2. old clang-tidy had `grep "cpp"` does this one grep only` hpp`?

Reply:
>     1. How this will work on master baranch?
> 
>     2. old clang-tidy had `grep "cpp"` does this one grep only` hpp`?

1. We will have the difference between HAED and the commit from last merge.
2. `grep "cpp"` was only used for the `filter_compile_commands.py` script. I use the same filter for the tidy svript. The purpose of the `filter_compile_commands.py` script is not clear to me. Because, It doesn't send any information to the tidy script, it just modify files in build directory.

Reply:
I believe `git diff $(git merge-base --fork-point $PR_BASE $PR_BRANCH)..$PR_BRANCH`  would be correct way to achieve what you want